src/Moryx.Resources.Management/Resources/ResourceManager.cs

```diff
@@ -113,26 +124,57 @@ public void Initialize()
                 // Create all objects
                 var allResources = ResourceEntityAccessor.FetchResourceTemplates(uow);
                 if (allResources.Count > 0)
-                {
                     LoadResources(allResources);
-                }                
             }
 
             _startup = ResourceStartupPhase.Initializing;
-            // Boot resources
-            Parallel.ForEach(Graph.GetAll(), resourceWrapper =>
-            {
-                try
-                {
-                    resourceWrapper.Initialize();
-                }
-                catch (Exception e)
-                {
-                    resourceWrapper.ErrorOccured();
-                    Logger.Log(LogLevel.Warning, e, "Failed to initialize resource {0}-{1}", resourceWrapper.Target.Id, resourceWrapper.Target.Name);
-                }
-            });
+            // initialize resources
+            Parallel.ForEach(Graph.GetAll(), InitializeResource);
             _startup = ResourceStartupPhase.Initialized;
+
+            // start resources
+            _startup = ResourceStartupPhase.Starting;
+            Parallel.ForEach(Graph.GetAll(), StartResource);
+            _startup = ResourceStartupPhase.Started;
+        }
+
+        private void ResourceFailed(Resource resource, Exception e)
+        {
+            //populate the failed resources list for tracking
+            lock (_failedResources)
+            {
+                if(_runningResources.Any(x => x.Id == resource.Id))
+                    _runningResources.Remove(resource);
+
+                if(!_failedResources.Any(x => x.Id == resource.Id))
+                _failedResources.Add(resource);
```

Comment:
Resources can not switch between the two and iterating the entire list (`Any`) for every resource will kill boot times for large applications (>5000 resources).